File: src/main/java/com/server/ordering/domain/MyCoupon.java
package com.server.ordering.domain;

import com.server.ordering.domain.member.Customer;
import lombok.*;

import javax.persistence.*;

import static javax.persistence.FetchType.*;
import static lombok.AccessLevel.*;

@Entity
@Getter
@RequiredArgsConstructor
@NoArgsConstructor(access = PROTECTED)
public class MyCoupon {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "my_coupon_id")
    private Long id;

    @NonNull
    private String serialNumber;
    @NonNull
    private int value;
    @NonNull
    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "customer_id")
    private Customer customer;
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/MyCoupon.kt
package com.byeoru.ordering_server.domain

import com.byeoru.ordering_server.domain.member.Customer
import javax.persistence.*

@Entity
class MyCoupon(serialNumber: String,
               value: Int,
               customer: Customer) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "my_coupon_id")
     var id: Long? = null
         protected set

     var serialNumber: String = serialNumber
         protected set

     var value = value
         protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id")
     var customer: Customer = customer
         protected set
}